GitUI/BranchTreePanel/RepoObjectsTree.ContextActions.cs

```diff
@@ -37,91 +37,85 @@ partial class RepoObjectsTree : IMenuItemFactory
         /// </summary>
         private MenuItemsGenerator<TagNode> _tagNodeMenuItems;
 
-        private void ContextMenuAddExpandCollapseTree(ContextMenuStrip contextMenu)
+        /// <summary>Toggles the <paramref name="items"/> <see cref="ToolStripItem.Visible"/>
+        /// as well as <see cref="ToolStripItem.Enabled"/> properties depending on <paramref name="enabled"/>.
+        /// Prefer this over only toggling the visibility of an item to enable determining whether the context menu will (once open)
+        /// contain any visible items via <see cref="ToolStripItem.Enabled"/> in <see cref="contextMenu_Opening(object, CancelEventArgs)"/>
+        /// even before the menu itself (as the visual parent) is visble and <see cref="ToolStripItem.Visible"/> of any item therefore false.</summary>
+        private void ToggleMenuItems(bool enabled, params ToolStripItem[] items) => items.ForEach(item => item.Visible = item.Enabled = enabled);
+
+        private void ToggleMenuItems<TNode>(MenuItemsGenerator<TNode> generator, Func<ToolStripItemWithKey, bool> isEnabled) where TNode : class, INode
+            => generator.ForEach(i => ToggleMenuItems(isEnabled(i), i.Item));
+
+        /* add Expand All / Collapse All menu entry
+         * depending on whether node is expanded or collapsed and has child nodes at all */
+        private void ToggleExpandCollapseContextMenu(ContextMenuStrip contextMenu, NodeBase[] selectedNodes)
         {
-            // Add the following to the every participating context menu:
-            //
-            //    ---------
-            //    Collapse All
-            //    Expand All
+            var multiSelectedParents = selectedNodes.HavingChildren().ToArray();
 
-            Tree? treeNode = (contextMenu.SourceControl as TreeView)?.SelectedNode?.Tag as Tree;
-
-            if (contextMenu == menuMain)
+            // selection contains nodes with children
+            if (multiSelectedParents.Length > 0)
             {
-                contextMenu.Items.Clear();
-                contextMenu.Items.Add(mnubtnCollapse);
-                contextMenu.Items.Add(mnubtnExpand);
-                if (treeNode is not null)
+                if (contextMenu == menuMain)
                 {
-                    AddMoveUpDownMenuItems();
+                    contextMenu.Items.Clear();
```

Comment:
This should not be necessary anymore (see above). Particularly, it should not been hidden in this - now specific - function.

Reply:
The clearing of the menu items here is a symptom of how the context menu creation is implemented. I'll see if I can get rid of it when I fix the user script item order, but can't guarantee that.

Reply:
According to the comment above, only the menu items for scripts might need to be removed from `Items`.
(If a `Clear` would be necessary, this generic functionality must not be hidden in a function dedicated to a few specific menu items. Most devs would be surprised that `ToggleExpandCollapseContextMenu` deletes all items.)